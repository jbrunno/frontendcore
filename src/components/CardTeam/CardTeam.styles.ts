import { css, styled } from '@mui/material/styles';
import Box from '../Box/Box';
import Card from '../Card/Card';

export const CardStyled = styled(Card)(
  ({ theme }) => css`
    display: grid;
    grid-template-columns: auto 1fr auto;
    align-items: center;
    grid-gap: ${theme.fn.spacing(16)};
    padding: ${theme.fn.spacing(16)};
  `,
);

type IconBoxProps = {
  backgroundBox?: string;
};

export const IconBoxStyled = styled('div')<IconBoxProps>(
  ({ theme, backgroundBox }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: ${theme.fn.spacing(8)};
    background: ${backgroundBox};
  `,
);

export const BoxStyled = styled(Box)`
  display: grid;
  grid-template-rows: auto auto auto;
`;
