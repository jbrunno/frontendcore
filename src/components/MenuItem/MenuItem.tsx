import React from 'react';
import {
  MenuItem as MuiMenuItem,
  MenuItemProps as MuiMenuItemProps,
} from '@mui/material';

export type MenuItemProps = MuiMenuItemProps;

export function MenuItem(props: MenuItemProps) {
  return <MuiMenuItem {...props} />;
}
export default MenuItem;
