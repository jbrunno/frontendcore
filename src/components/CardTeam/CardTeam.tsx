import React, { ReactNode } from 'react';
import {
  MoreVert as MoreVertIcon,
  Group as GroupIcon,
} from '@mui/icons-material';
import { useTheme } from '@mui/material/styles';
import { BoxStyled, CardStyled, IconBoxStyled } from './CardTeam.styles';
import { Typography } from '../Typography/Typography';
import Tooltip from '../Tooltip/Tooltip';
import IconButton from '../IconButton/IconButton';
import { useCardTeam } from './hooks/useCardTeam';
import Menu from '../Menu/Menu';

type CardTeamProps = {
  className?: string;
  nome: string;
  supervisor?: string;
  totalUsers: string;
  colorIcon?:
    | 'inherit'
    | 'disabled'
    | 'action'
    | 'primary'
    | 'secondary'
    | 'error';
  backgroundBox?: string;
  tooltipTitle?: string;
  menuItems?: ReactNode;
};

export function CardTeam({
  className,
  nome,
  supervisor,
  totalUsers,
  colorIcon = 'primary',
  backgroundBox,
  tooltipTitle = 'Opções',
  menuItems,
}: CardTeamProps) {
  const {
    data: { menuOpen, anchorEl },
    actions: { handleMenuOpen, handleClose },
  } = useCardTeam();

  const theme = useTheme();

  return (
    <CardStyled className={className}>
      <IconBoxStyled backgroundBox={backgroundBox || theme.palette.primary[50]}>
        <GroupIcon color={colorIcon} fontSize="large" />
      </IconBoxStyled>
      <BoxStyled>
        <Typography variant="h5">{nome}</Typography>
        <Typography color="textSecondary" variant="caption">
          {supervisor}
        </Typography>
        <Typography color="textSecondary" variant="caption">
          {totalUsers} usuários
        </Typography>
      </BoxStyled>
      <Tooltip title={tooltipTitle} arrow>
        <IconButton aria-label="settings" onClick={handleMenuOpen}>
          <MoreVertIcon />
        </IconButton>
      </Tooltip>
      {menuItems && (
        <Menu open={menuOpen} onClose={handleClose} anchorEl={anchorEl}>
          {menuItems}
        </Menu>
      )}
    </CardStyled>
  );
}

export default CardTeam;
