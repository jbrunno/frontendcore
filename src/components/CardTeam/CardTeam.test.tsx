import React from 'react';
import { render } from '@test-utils/component-testing';
import { CardTeam } from './CardTeam';
import MenuItem from '../MenuItem/MenuItem';

test('renders the CardTeam', () => {
  const { container } = render(
    <CardTeam
      className="card-team"
      nome="Nome"
      supervisor="Supervisor"
      totalUsers="123"
      colorIcon="primary"
      backgroundBox="#F1E8FF"
      tooltipTitle="Tooltip Title"
      menuItems={[
        <MenuItem>Adicionar Administrador</MenuItem>,
        <MenuItem>Alterar Nome</MenuItem>,
        <MenuItem>Alterar LÃ­der</MenuItem>,
        <MenuItem>Excluir Equipe</MenuItem>,
        <MenuItem>Alterar Meta</MenuItem>,
      ]}
    />,
  );
  expect(container).toMatchSnapshot();
});
