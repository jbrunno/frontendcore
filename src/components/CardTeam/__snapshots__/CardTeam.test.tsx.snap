// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CardTeam 1`] = `
.emotion-2 {
  background-color: #fff;
  color: #3B3B41DE;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  background-color: #fff;
  display: grid;
  grid-template-columns: auto 1fr auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-gap: 1rem;
  padding: 1rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  background: #F1E8FF;
}

.emotion-5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 2.1875rem;
  color: #4A31FF;
}

.emotion-7 {
  display: grid;
  grid-template-rows: auto auto auto;
}

.emotion-8 {
  margin: 0;
  font-weight: 700;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0;
  font-family: Lato,sans-serif;
}

.emotion-9 {
  margin: 0;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.125rem;
  letter-spacing: 0.0106rem;
  font-family: Lato,sans-serif;
  color: #3B3B4199;
}

.emotion-11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: #0000008A;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-11::-moz-focus-inner {
  border-style: none;
}

.emotion-11.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-11 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-11:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-11:hover {
    background-color: transparent;
  }
}

.emotion-11.Mui-disabled {
  background-color: transparent;
  color: #00000042;
}

.emotion-12 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-13 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card-team emotion-0 emotion-1 emotion-2"
  >
    <div
      class="emotion-3 emotion-4"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge emotion-5"
        data-testid="GroupIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
        />
      </svg>
    </div>
    <div
      class="emotion-6 MuiBox-root emotion-7"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 emotion-8"
      >
        Nome
      </h5>
      <span
        class="MuiTypography-root MuiTypography-caption emotion-9"
      >
        Supervisor
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption emotion-9"
      >
        123
         usu√°rios
      </span>
    </div>
    <div
      aria-label="Tooltip Title"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        aria-label="settings"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-11"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-12"
          data-testid="MoreVertIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root emotion-13"
        />
      </button>
    </div>
  </div>
</div>
`;
